core/tests/protractor_utils/waitFor.js

```diff
@@ -181,6 +181,33 @@ var fileToBeDownloaded = async function(filename) {
   }, DEFAULT_WAIT_TIME_MSECS, 'File was not downloaded!');
 };
 
+var clientSideRedirection = async function(action, check) {
+  // Client side redirection is known to cause "both angularJS testability
+  // and angular testability are undefined" flake.
+  // As suggested by protractor devs here (http://git.io/v4gXM), waiting for
+  // angular is disabled during client side redirects.
+  await browser.waitForAngularEnabled(false);
+
+  // Action triggering redirection.
+  await action();
+
+  // The action only triggers the redirection but does not wait for it to
+  // complete. Manually waiting for redirection here.
+  await browser.driver.wait(async() => {
+    var url = await browser.driver.getCurrentUrl();
+    // Condition to wait on.
+    return check(url);
```

Comment:
Note that we already have a function to wait for url redirection:

https://github.com/oppia/oppia/blob/37c24c46e6fc5a9ee1e8df6fb62d55737fa1ced3/core/tests/protractor_utils/waitFor.js#L140-L144

It's not as generic as your approach though, so I think your approach is fine.